libvast/vast/concept/parseable/vast/data.hpp

```diff
@@ -48,9 +48,15 @@ struct data_parser : parser<data_parser> {
     auto ws = ignore(*parsers::space);
     auto x = ws >> ref(p) >> ws;
     auto kvp = x >> "->" >> x;
-    auto field = parsers::identifier >> ":" >> x;
     auto trailing_comma = ~(',' >> ws);
+    auto field = ws >> parsers::identifier >> ":" >> x;
     // clang-format off
+    auto anon_field = x ->* [](data value) {
+      return std::make_pair(std::string{}, std::move(value));
+    };
+    auto record_parser =
+        '<' >> ~as<record>(field % ',') >> trailing_comma >> '>'
+      | '<' >> ~as<record>(anon_field % ',') >> trailing_comma >> '>';
```

Comment:
```suggestion
    auto field = named_field | anon_field;
    auto record_parser = '<' >> ~as<record>(field % ',') >> trailing_comma >> '>'
```

Reply:
That does not make sense to me. It would allow records like `<_, orig_h: 1.2.3.4>`, but then we'd need to deal with those ambiguities somehow.

Reply:
Good point. Could you write a comment that we deliberately went with *either* anonymous *or* named fields?